### PR TITLE
Add `related_playlist` to Account and Channel

### LINKED ISSUE #218: Where params aren't respected for playlists
I believe we'd need something along these lines: https://github.com/Fullscreen/yt/blob/master/lib/yt/collections/channels.rb#L24-L28

### PR DESCRIPTION
**Pull request by claudiofullscreen**
With related_playlists, you can retrieve special playlists associated
to a channel such as "Liked videos" or "Uploads"

Closes #218

### CODE REVIEW

**Comment by claudiofullscreen** [comment]
@trevorturk This should work to close #218 

I need to warn you that YouTube has deprecated "Favorites". 
Therefore _old_ YouTube channels will have a related playlist called "Favorites", but new accounts won’t.
Every account should have the "Liked videos" playlist, though.

**Comment by trevorturk** [comment]
This looks great! Would you like me to test things out now, or should I wait until you merge this into the master branch? 

Thanks for the warning about the deprecated "Favorites" playlist. Actually, I'm finding that things are quite odd even with new accounts. For example, here's a new test account that I made, which has 3 different playlists returned by the API with various calls:

First, the "Favorites" playlist for user (returned by relatedPlaylists) does not exist when requested via the website when logged in as the user: https://www.youtube.com/playlist?list=FLIbyQMQMIm3HAbJ1IXaOFaQ

Second, the "Liked videos" playlist for user (returned by relatedPlaylists) is private: https://www.youtube.com/playlist?list=LLIbyQMQMIm3HAbJ1IXaOFaQ 

Third, this oddball "Favorites" playlist for the channel (returned as the only playlist for the channel when requested via the API) seems to combine the above "Favorites" and "Liked videos" even though the "Favorites" playlist from relatedPlaylists allegedly doesn't exist. (Videos appear in this playlist if you use the "Add to" menu and select the allegedly nonexistent "Favorites" playlist _or_ if click the "Like" / thumbs-up button on any video: https://www.youtube.com/playlist?list=PLY84RbNn14ftXWknvOGm95Z6qjkJi-Qpl 

In my use case, I want to collect all favorites/likes for a given user. So, with older accounts I'm able to use the single relatedPlaylists call to get the (valid) playlist ids for "Favorites" and "Liked videos" and then I can collect the videos from those two lists. For newer accounts that have two different "Favorites" playlist ids (one gotten from relatedPlaylists and one gotten from the channel's playlists) I collect videos from the channel's "Favorites" playlist, which mysteriously combines the two separate playlists into a single combined playlist. 

Sorry, I know that's confusing, but anyway I thought it worth trying to explain :)

**Comment by claudiofullscreen** [comment]
Hello @trevorturk !

My first suggestion is to stop using "Favorites" at all.
YouTube deprecated "Favorites". They might still be around for old users, but they will be gone soon. 

I don't seem to find the blog post where they announced it, but [this link](https://developers.google.com/youtube/analytics/v1/dimsmets/mets#favoritesAdded) should work.

I also noticed that, when you ask for your own "related playlists", YouTube includes some IDs that do not exist at all. I believe this is a YouTube bug and I will report this to YouTube.

However, I have already taken care of this weirdness in this PR by eager-loading the `Yt::Playlist`, not just their IDs. Therefore, I invite you to try your code against this branch. You should see that those "unknown" playlists **are not included** in `account.related_playlists` at all.

Regarding the fact that the "Liked Videos" playlist for a user is private, there's nothing I can do about it. Some users have this playlist set to public (such as [this one](https://www.youtube.com/playlist?list=LLxO1tY8h1AhOz0T4ENwmpow)), other users have set it to private and… looks like YouTube does not let you change the privacy status of this playlist anymore!

Summing up, I think this PR might still help you retrieve _some data_ for _some YouTube users_. Better than nothing, but it won't solve YouTube bugs or weirdnesses. Give it a try and let me know.

**Comment by trevorturk** [comment]
That's great about the eager-loading. I'll give this branch a try now and report back.

**Comment by trevorturk** [comment]
I just tested with my small sample set of accounts and it looks great! Thank you!

**Comment by claudiofullscreen** [comment]
@trevorturk Update to `0.25.4` and you should be good to go!

**Comment by trevorturk** [comment]
Awesome! Thank you <3
